src/roaring_treemap/iter.rs

```diff
@@ -0,0 +1,181 @@
+use std::collections::BTreeMap;
+use std::collections::btree_map;
+use std::iter::{self, FromIterator};
+
+use iter::Iter as Iter32;
+use iter::IntoIter as IntoIter32;
+use super::util;
+use RoaringBitmap;
+use RoaringTreemap;
+
+struct To64Iter<'a> {
+    hi: u32,
+    inner: Iter32<'a>,
+}
+
+impl<'a> Iterator for To64Iter<'a> {
+    type Item = u64;
+    fn next(&mut self) -> Option<u64> {
+        self.inner.next().map(|n| util::join(self.hi, n))
+    }
+}
+
+fn to64iter<'a>(t: (&'a u32, &'a RoaringBitmap)) -> To64Iter<'a> {
+    To64Iter {
+        hi: *t.0,
+        inner: t.1.iter(),
+    }
+}
+
+struct To64IntoIter {
+    hi: u32,
+    inner: IntoIter32,
+}
+
+impl Iterator for To64IntoIter {
+    type Item = u64;
+    fn next(&mut self) -> Option<u64> {
+        self.inner.next().map(|n| util::join(self.hi, n))
+    }
+}
+
+fn to64intoiter(t: (u32, RoaringBitmap)) -> To64IntoIter {
+    To64IntoIter {
+        hi: t.0,
+        inner: t.1.into_iter(),
+    }
+}
+
+type InnerIter<'a> = iter::FlatMap<btree_map::Iter<'a, u32, RoaringBitmap>,
+                                   To64Iter<'a>,
+                                   fn((&'a u32, &'a RoaringBitmap)) -> To64Iter<'a>>;
+type InnerIntoIter = iter::FlatMap<btree_map::IntoIter<u32, RoaringBitmap>,
+                                   To64IntoIter,
+                                   fn((u32, RoaringBitmap)) -> To64IntoIter>;
+
+/// An iterator for `RoaringTreemap`.
+pub struct Iter<'a> {
+    inner: InnerIter<'a>,
+    size_hint: u64,
+}
+
+/// An iterator for `RoaringTreemap`.
+pub struct IntoIter {
```

Comment:
I think you're right.  I've decided to expose them at roaring_treemap::{Iter, IntoIter} by putting a `pub use` in roaring_treemap/mod.rs.  That way, it follows the model of std::collections::BTreeMap and std::collections::btree_map::Iter.  Does that sound good?  Maybe I should just keep RoaringTreemap in the module as well?

Don't you currently have the same problem with your IntoIter for RoaringBitmap?  I don't see any way it's exposed, but you're using it when you `impl IntoIterator for RoaringBitmap`.  I don't understand how that would work or how the documentation would work.

Ok, I just ran cargo doc, and yeah, it doesn't work.